File: Dynamic-Memory-Allocation/C programs/modifySize.c
// 10. Write a C program to modify the size of an array and utilize that during run time.

#include <stdio.h>
#include <stdlib.h>

int main()
{
    int *arr, n, m;
    printf("Enter size of the array: ");
    scanf("%d", &n);
    arr = (int *)malloc(n * sizeof(int));

    for (int i = 0; i < n; i++)
    {
        printf("Enter Array[%d]: ", i);
        scanf("%d", arr + i);
    }

    printf("\nHow many extra elements you want to add: ");
    scanf("%d", &m);
    arr = (int *)realloc(arr, sizeof(int) * (m + n));

    for (int i = n; i < m + n; i++)
    {
        printf("Enter Array[%d]: ", i);
        scanf("%d", arr + i);
    }

    printf("\nNew array after reallocation is\n");
    for (int i = 0; i < m + n; i++)
    {
        printf("%d ", *(arr + i));
    }

    free(arr);
    return 0;
}

// OUTPUT
// Enter size of the array: 2
// Enter Array[0]: 2
// Enter Array[1]: 4

// How many extra elements you want to add: 2
// Enter Array[2]: 6
// Enter Array[3]: 8

// New array after reallocation is
// 2 4 6 8

File: Dynamic-Memory-Allocation/Python programs/modifySize.py
# 10. Write a python program to modify the size
# of an array and utilize that during run time.

arr = [1, 2, 3, 4]
n = int(input("Enter number of extra elements: "))
print("Enter the elements: ")
for i in range(n):
    ele = int(input())
    arr.append(ele)
print(arr)

# OUTPUT
# Enter number of extra elements: 3
# Enter the elements:
# 9
# 8
# 7
# [1, 2, 3, 4, 9, 8, 7]
